src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -22,44 +26,54 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
      * actions on the existing resource if it exists or create it if it doesn't.
      *
      * @param resourceDrafts the list of new resources as drafts.
-     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link V} which
+     *      is a subclass of {@link BaseSyncStatisticsBuilder} representing the builder of {@code statistics} baked for
+     *      the sync process
      */
-    protected abstract CompletionStage<U> process(@Nonnull final List<T> resourceDrafts);
+    protected abstract CompletionStage<V> process(@Nonnull final List<T> resourceDrafts);
 
 
     /**
      * Given a list of resource (e.g. categories, products, etc..) drafts. This method compares each new resource in
      * this list with it's corresponding old resource in a given CTP project, and in turn it either issues update
      * actions on the existing resource if it exists or create it if it doesn't.
      *
-     * <p>The time before and after the actual sync process starts is recorded in the {@link BaseSyncStatistics}
-     * container so that the total processing time is computed in the statistics.
-     *
      * @param resourceDrafts the list of new resources as drafts.
      * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} of this sync process.
      */
     public CompletionStage<U> sync(@Nonnull final List<T> resourceDrafts) {
-        statistics.startTimer();
-        return process(resourceDrafts).thenApply(resultingStatistics -> {
-            resultingStatistics.calculateProcessingTime();
+        final long startTime = System.currentTimeMillis();
+        return process(resourceDrafts).thenApply(resultingStatisticsBuilder -> {
+            resultingStatisticsBuilder.setProcessingTimeInMillis(System.currentTimeMillis() - startTime);
+            final U resultingStatistics = resultingStatisticsBuilder.build();
```

Comment:
But 2 objects have to be built then (one for joining to instance statistics and the second for return statement)